File: Gui/Linn/Gui/SceneGraph/RenderState.py
from Ft.Xml import XPath
import Linn.Gui.Serialize
import Linn.Gui.Resources.TextureManager

class RenderState(Linn.Gui.Serialize.ISerialize):
    def __init__(self, aTexture=None):
        self.iTexture = aTexture
        
    def Texture(self):
        return self.iTexture
    
    def SetTexture(self, aTexture):
        self.iTexture = aTexture
        
    def Load(self, aStream):
        texture = aStream.LoadNode('Texture')
        if len(texture) != 1:
            raise Linn.Gui.Serialize.InvalidLayoutFile('<Texture>', len(texture), 1)
        self.iTexture = texture[0].firstChild.nodeValue
        
    def Save(self, aStream):
        rs = aStream.CreateElement('RenderState')
        if self.iTexture:
            texture = aStream.CreateElement('Texture')
            texture.appendChild(aStream.CreateTextNode(self.iTexture.Filename()))
            rs.appendChild(texture)
        aStream.SaveNode(rs)
        
    def Link(self):
        #print 'Linking [RenderState]'
        if self.iTexture:
            self.iTexture = Linn.Gui.Resources.TextureManager.GetTextureManager().TextureByNameOrLoad(self.iTexture)
            
            

File: Public/Gui/Linn/Gui/SceneGraph/RenderState.cs
using Linn.Gui.Resources;
using System.Xml.XPath;
using System.Xml;

namespace Linn {
namespace Gui {

public class RenderState : ISerialiseObject
{
    public void Load(XmlNode aXmlNode) {
        XmlNodeList list;
        list = aXmlNode.SelectNodes("Texture");
        if(list != null) {
            if(list.Count > 0) {
                Assert.Check(list.Count == 1);
                iTexture = new ReferenceTexture(list[0].FirstChild.Value);
            }
        }
    }
    
    public void Link() {
        iTexture.Link();
    }
    
    public void Save(XmlTextWriter aWriter) {
        aWriter.WriteStartElement("RenderState");
        if(iTexture.Object != null) {
            aWriter.WriteStartElement("Texture");
            aWriter.WriteString(iTexture.Object.Filename);
            aWriter.WriteEndElement();
        }
        aWriter.WriteEndElement();
    }
    
    public ReferenceTexture Texture {
        get {
            return iTexture;
        }
        set {
            iTexture = value;
        }
    }
    
    private ReferenceTexture iTexture = new ReferenceTexture();
}
    
} // Gui
} // Linn
